enterprise/core-edge/src/test/java/org/neo4j/coreedge/catchup/storecopy/LocalDatabaseTest.java

```diff
@@ -96,6 +99,92 @@ public void shouldThrowWhenDifferentStoreIds() throws Throwable
         }
     }
 
+    @Test
+    public void storeCopyShouldResetStoreId() throws Throwable
+    {
+        // given
+        NeoStoreDataSource mockDatasource = mock( NeoStoreDataSource.class );
+        when( mockDatasource.getStoreId() ).thenReturn( new org.neo4j.kernel.impl.store.StoreId( 1, 2, 3, 4, 5 ) );
+        StoreId copied = new StoreId( 5, 6, 7, 8 );
+        LocalDatabase localDatabase = createLocalDatabase( mockDatasource );
+        MemberId memberId = mock( MemberId.class );
+        StoreFetcher storeFetcher = mock( StoreFetcher.class );
+        when( storeFetcher.storeId( memberId ) ).thenReturn( copied );
+
+        // when
+        // a local database is created with the initial store id
+        localDatabase.start();
+
+        // and stopped, the store copied, and restarted
+        localDatabase.stop();
+        localDatabase.copyStoreFrom( memberId, storeFetcher );
+        when( mockDatasource.getStoreId() ).thenReturn( new org.neo4j.kernel.impl.store.StoreId( 5, 6, 3, 7, 8 ) );
+        localDatabase.start();
+
+        // then
+        // the returned storeId should be the one of the copied store
+        assertEquals( copied, localDatabase.storeId() );
+    }
+
+    @Test
+    public void beingStoppedShouldReturnNonMatchingStoreId() throws Throwable
+    {
+        // given
+        LocalDatabase localDatabase = createLocalDatabase( new org.neo4j.kernel.impl.store.StoreId( 1, 2, 3, 4, 5 ) );
+
+        // when
+        // a local database is created with the initial store id
+        localDatabase.start();
+
+        // and stopped
+        localDatabase.stop();
+
+        // then
+        // the storeId returned in this state should be the default store id
+        assertEquals( StoreId.DEFAULT, localDatabase.storeId() );
```

Comment:
This equality check I assume passes because it is the exact same object, even though DEFAULT is defined to "never" be equal. Slightly confusing.